addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -45,7 +45,40 @@ import { jsonrpc } from "@web/core/network/rpc_service";
 const preserveCursor = OdooEditorLib.preserveCursor;
 const { DateTime } = luxon;
 const resetOuids = OdooEditorLib.resetOuids;
-
+let _rpcCache = {};
+const clearRpcCache = () => {
+    _rpcCache = {};
+};
+/**
+ * Caches rpc service
+ * @param {Function} rpc_service 
+ * @param  {...any} args 
+ * @returns 
+ */
+async function rpcCached(rpc_service, ...args) {
+    const cacheId = JSON.stringify(...args);
+    if (!_rpcCache[cacheId]) {
+        _rpcCache[cacheId] = rpc_service(...args);
+    }
+    return _rpcCache[cacheId];
+}
+/**
+ * Caches orm service
+ * @param {Function} orm 
+ * @returns 
+ */
+function ormCached(orm) {
+    const cache = _rpcCache;
+    return Object.assign(Object.create(orm), {
+        call() {
+            const cacheId = JSON.stringify(arguments);
+            if (!cache[cacheId]) {
+                cache[cacheId] = orm.call(...arguments);
+            }
+            return cache[cacheId];
+        },
+    });
+}
```

Comment:
Arf, did not know that the m2o one was replaced to use the ORM service :grimacing: Your adaptation seems fine but it's strange to have two different implementation for the same name: one does the call using the given service (rpcCached) and one patches the given service for future calls (ormCached). Choose one IMO :wink: 

Also, maybe I am just wrong: I had in mind than the m2o implementation used this._rpc too (as it was the case at the beginning). But `JSON.stringify(arguments)` is not using the same format for rpc or orm services, maybe it should be the same "clearRpcCache" for both clearing rpc service and m2o service but they should use different cache objects.

Reply:
I'm gonna try. Thanks

Reply:
The implementation has been improved, and I have successfully resolved a linter issue related to the hasOwnProperty method. Take a look.